src/main/java/org/jabref/logic/importer/fetcher/DoiResolution.java

```diff
@@ -33,7 +33,10 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(DoiResolution.class);
 
     /**
-     * Hosts for which tailored fetchers exist, so this fetcher is not needed.
+     * A DOI leads to the page of the paper. We assume that there is the PDF of the paper.
+     * Thus, the DOI fetcher is ranked highest (see {@link TrustLevel}).
+     * Some publishers do not publish the PDF of the paper, but the full proceedings on the page
+     * of the paper. For these publishers, we need to skip the DOI search.
```

Comment:
This is not entirely correct AFAICT. The problem with getting proceedings instead of the individual paper exists only for Springer. 

Other publishers (such as IEEE) can be added here if a tailored fetcher for them exists that also fetches a PDF for a DOI. Such a tailored fetcher will run in parallel with this class here and get at least the same result, if not a better one, but with a lower TrustLevel (like for Springer DOIs). Therefore, if the DOI at hand leads to one of those publishers, we can skip the DOI search via this class here.

Reply:
We assume that the DOI fetcher returns a DOI if it is (nearly) 100% sure that a DOI is determined. We assume the custom fetchers also return something even if they are not 100% sure (right @stefan-kolb?)

This is what I tried to write down at https://github.com/JabRef/jabref/pull/6990/files#diff-d3fe05c852562027c0866670547c8b8a

If we trust in general ALL custom fetchers more, we need to switch the numbers for `SOURCE` and `PUBLSIHER` in `TrustLevel` (https://github.com/JabRef/jabref/pull/6990/files#diff-77f5ba02dd7375d532d9c3aee3527dfc)

Reply:
We fixed a similarity problem inside the DOIResolution that REALLY caused this issue.